From c9a24d9bc6f06bb7158dc00feba09b0f938daef3 Mon Sep 17 00:00:00 2001
From: Thieu Le <thieule@chromium.org>
Date: Tue, 26 Jul 2011 16:15:10 -0700
Subject: [PATCH 79/87] ecryptfs: Make inode bdi consistent with superblock
 bdi

Make the inode mapping bdi consistent with the superblock bdi so that
dirty pages are flushed properly.

Signed-off-by: Thieu Le <thieule@chromium.org>
Cc: <stable@kernel.org> [2.6.39+]
Signed-off-by: Tyler Hicks <tyhicks@linux.vnet.ibm.com>
---
 fs/ecryptfs/inode.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/fs/ecryptfs/inode.c
+++ b/fs/ecryptfs/inode.c
@@ -69,6 +69,7 @@ static int ecryptfs_inode_set(struct ino
 	inode->i_ino = lower_inode->i_ino;
 	inode->i_version++;
 	inode->i_mapping->a_ops = &ecryptfs_aops;
+	inode->i_mapping->backing_dev_info = inode->i_sb->s_bdi;
 
 	if (S_ISLNK(inode->i_mode))
 		inode->i_op = &ecryptfs_symlink_iops;
